### DISCUSSION TITLE
Why JanusGraphServer can't start

### DISCUSSION THREAD

**Discussion by FantasticCode2019** [JanusGraph/janusgraph#3180] [Q&A]
I have used this package(janusgraph-full-0.6.2.zip).The JanusGraph application runs on a virtual machine.But the JVM can't allocate memory.Please give some suggestion and Thanks for you.
`Java HotSpot(TM) 64-Bit Server VM warning: INFO: os::commit_memory(0x00000006c5330000, 4207738880, 0) failed; error='Cannot allocate memory' (errno=12)`
`There is insufficient memory for the Java Runtime Environment to continue.`
`Native memory allocation (mmap) failed to map 4207738880 bytes for committing reserved memory.`

**Comment by vtslab** [comment]
Some questions:
- What command(s) do you issue to "run the JanusGraph application"? If you use bin/janusgraph.sh can you please show the full terminal log.
- What memory does your VM have available? The 4GB in the error log is about the default max heap size of the JVM.

**Reply by FantasticCode2019** [reply]
thanks,I get it.I solved this problem by reducing the JVM's default max stack size.
